easycv/predictors/detector.py

```diff
@@ -252,6 +261,153 @@ def predict(self, input_data_list, batch_size=-1, to_numpy=True):
         return output_list
 
 
+@PREDICTORS.register_module()
+class TorchViTDetPredictor(PredictorInterface):
+
+    def __init__(self, config, checkpoint):
+
+        self.device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        if config.startswith('http'):
+
+            r = requests.get(config)
+            # download config in current dir
+            tpath = config.split('/')[-1]
+            while not osp.exists(tpath):
+                try:
+                    with open(tpath, 'wb') as code:
+                        code.write(r.content)
+                except:
+                    pass
+
+            config = tpath
+
+        self.cfg = mmcv_config_fromfile(config)
+        self.CLASSES = self.cfg.CLASSES
+        # dynamic adapt mmdet models
+        dynamic_adapt_for_mmlab(self.cfg)
+        self.model = build_model(self.cfg.model)
+        checkpoint = load_checkpoint(
+            self.model, checkpoint, map_location='cpu')
+        self.model.to(self.device)
+        self.model.eval()
+
+    def inference_detector(self, imgs):
```

Comment:
PredictorInterface use predict function to do inference

Reply:
done